### DISCUSSION TITLE
Workspaces List

### DISCUSSION THREAD

**Discussion by OmerGronich** [e-square-io/angular-cli-gui#43] [High Level Design]
**Overview and Goals**

- View a list of connected workspaces
- Favorite a connected workspace
- Rename a workspace
- Delete a workspace
- Search a workspace

---

**Architecture and Design**

**TDB** List of workspaces will either be saved to localStorage/indexDb or a local json file

---

**Data Model**

```tsx
interface Workspace {
  name: string;
  path: string;
  isFavorite: boolean;
}
```

---

**User Interface Design**:

Reference: 

<img width="1728" alt="Untitled (1)" src="https://user-images.githubusercontent.com/59396491/220064537-66bc39d0-cd1a-4ebd-8964-e420dee11395.png">

**Comment by Danieliverant** [comment]
Workspace if for Project also? maybe we want to distinguish between project and workspace on the modal level?

**Reply by ronnetzer** [reply]
what is the difference?

**Reply by Danieliverant** [reply]
The difference is in the UI level, functionally you don't have a difference, but maybe on this screen we want to indicate (by some icon) that "this is a Workspace" and "this is a Project" and in the next screen (after picking one) you will need to show/not to show the "projects" dropdown to know on what project to execute the command

**Reply by OmerGBlink** [reply]
I don't think we need to show projects in this screen it will just cause confusion. This domain is all about managing workspaces. 
I think in all other screens we should allow the option to choose projects from the available projects in the specific connected workspace that is currently active

**Reply by Danieliverant** [reply]
@OmerGBlink so how can load a project that is not in a workspace?

**Reply by OmerGronich** [reply]
You can't , every project has to have a workspace @agroupp correct me if I'm wrong

**Reply by Danieliverant** [reply]
@OmerGronich @agroupp I don't get it - what should people that use plain Angular do?
